src/viewer.js

```diff
@@ -418,20 +434,66 @@ $.Viewer = function( options ) {
         maxImageCacheCount: this.maxImageCacheCount
     });
 
-    // Create the drawer
-    this.drawer = new $.Drawer({
-        viewer:             this,
-        viewport:           this.viewport,
-        element:            this.canvas,
-        debugGridColor:     this.debugGridColor
-    });
+    //Create the drawer based on selected options
+    if (Object.prototype.hasOwnProperty.call(this.drawerOptions, 'useCanvas') ){
+        $.console.error('useCanvas is deprecated, use the "drawer" option to indicate preferred drawer(s)');
+
+        // for backwards compatibility, use HTMLDrawer if useCanvas is defined and is falsey
+        if (!this.drawerOptions.useCanvas){
+            this.drawer = $.HTMLDrawer;
+        }
+
+        delete this.drawerOptions.useCanvas;
+    }
+    let drawerCandidates = Array.isArray(this.drawer) ? this.drawer : [this.drawer];
+    if (drawerCandidates.length === 0){
+        drawerCandidates = [$.DEFAULT_SETTINGS.drawer].flat(); // ensure it is a list
+        $.console.warn('No valid drawers were selected. Using the default value.');
```

Comment:
Does this mean if the user doesn't specify a drawer they get a warning? I don't think that should be the case... Just going with the defaults is fine and shouldn't be warned against.

Reply:
If no drawer option is added this warning doesn't apply - the default will be a non-empty array. If an empty array is passed, in this will throw a warning, then use the defaults. If an invalid drawer (bad string, bad object) is passed, an error is eventually thrown when the drawer can't be created.

Reply:
Gotcha... Thanks for the clarification!